### PR TITLE
gh-111354: remove comparisons with enum values, variable reuse, unused imports in genobject.c

### PR DESCRIPTION
**Pull request by iritkatriel**
Tidies up genobject.c:

- remove unused imports
- inline a one-line function that is only called once
- do not reuse the ``result`` variable for the ``arg``
- remove comparisons with frame state enum values

Reviewing individual commits might be easier.

* Issue: gh-111354

### CODE REVIEW

File: Objects/genobject.c [hunk 1]
```diff
@@ -216,10 +206,12 @@ gen_send_ex2(PyGenObject *gen, PyObject *arg, PyObject **presult,
...
     }
 
-    assert(gen->gi_frame_state < FRAME_EXECUTING);
+    assert((gen->gi_frame_state == FRAME_CREATED) ||
+           FRAME_STATE_SUSPENDED(gen->gi_frame_state));
+
     /* Push arg onto the frame's value stack */
-    result = arg ? arg : Py_None;
-    _PyFrame_StackPush(frame, Py_NewRef(result));
+    PyObject *arg_obj = arg ? arg : Py_None;
```
Comment (markshannon) [inline Objects/genobject.c:213]: Would it make sense to require `arg` to be not `NULL`? This is a static function, so we know all the callers.
(Not necessarily in this PR, though)

File: Objects/genobject.c [hunk 2]
```diff
@@ -349,7 +341,7 @@ _PyGen_yf(PyGenObject *gen)
     if (gen->gi_frame_state == FRAME_SUSPENDED_YIELD_FROM) {
         _PyInterpreterFrame *frame = (_PyInterpreterFrame *)gen->gi_iframe;
         assert(is_resume(frame->instr_ptr));
-        assert((frame->instr_ptr->op.arg & RESUME_OPARG_LOCATION_MASK) >= RESUME_AFTER_YIELD_FROM);
+        assert(RESUME_WITH_SUBITERATOR(frame->instr_ptr->op.arg));
```
Comment (markshannon) [inline Objects/genobject.c]: I'd rather keep the expanded form here. Using a macro means the reader needs to go look up that macro.
Also if it is a predicate, it should be named as one. I.e. `IS_RESUME_WITH_SUBITERATOR`

File: Include/internal/pycore_frame.h [hunk 3]
```diff
@@ -45,6 +45,7 @@ typedef enum _framestate {
 } PyFrameState;
 
 #define FRAME_STATE_SUSPENDED(S) ((S) == FRAME_SUSPENDED || (S) == FRAME_SUSPENDED_YIELD_FROM)
+#define FRAME_STATE_CLOSED(S) ((S) >= FRAME_COMPLETED)
```
Comment (markshannon) [inline Include/internal/pycore_frame.h]: Does the state being `FRAME_COMPLETED` mean that it is closed?
Can't it be completed, but not closed?

**Review by markshannon — COMMENTED** [review]
A few comments

File: Include/internal/pycore_frame.h (hunk 3 above)
Comment (iritkatriel) [reply Include/internal/pycore_frame.h]: There is no CLOSED state. There is COMPLETED and CLEARED. CLOSED is a new category of both. Maybe FINISHED is a better name for this.

**Review by markshannon — APPROVED** [review]

**Pull request merged**